frontend/public/components/configmap-and-secret-data.jsx

```diff
@@ -29,17 +29,17 @@ export class SecretData extends React.PureComponent {
   }
 
   render() {
-    const { data } = this.props;
+    const { heading, data, type } = this.props;
     const { showSecret } = this.state;
     const dl = [];
     Object.keys(data || {}).sort().forEach(k => {
-      const value = window.atob(data[k]);
+      const value = type === 'route.openshift.io' ? data[k] : window.atob(data[k]);
```

Comment:
I'd rather not have a special check like this. We should generalize the component if we want to reuse it.

I think it's cleaner just not to use SecretData since it's only one field we want to mask.

Reply:
Also, this is still cluttered way to make a new array...(but that's a lingering comment).

Reply:
+1 for generalizing component.